gluon/compileapp.py

```diff
@@ -691,18 +686,18 @@ def run_view_in(environment):
     else:
         filename = pjoin(folder, 'views', view)
         if os.path.exists(path): # compiled views
-            x = view.replace('/', '_')
-            files = ['views_%s.pyc' % x]
+            x = view.replace('/', '.')
+            files = ['views.%s.pyc' % x]
```

Comment:
compiled views use ```_``` as separator. for example ```views_default_index.html```
https://github.com/web2py/web2py/blob/master/gluon/compileapp.py#L490